OpenRA.Game/Scripting/ScriptContext.cs

```diff
@@ -166,52 +165,61 @@ public sealed class ScriptContext : IDisposable
 			PlayerCommands = FilterCommands(world.Map.Rules.Actors[SystemActors.Player], knownPlayerCommands);
 
 			runtime.Globals["EngineDir"] = Platform.EngineDir;
-			runtime.DoBuffer(File.Open(Path.Combine(Platform.EngineDir, "lua", "scriptwrapper.lua"), FileMode.Open, FileAccess.Read).ReadAllText(), "scriptwrapper.lua").Dispose();
-			tick = (LuaFunction)runtime.Globals["Tick"];
+
+			// Sandboxing
+			var allowedFunctions = new string[]
+			{
+				"assert", "error", "ipairs", "next", "pairs",
+				"pcall", "select", "tonumber", "tostring", "type", "unpack", "xpcall",
+				"math", "string", "table"
+			};
+
+			foreach (var fieldName in runtime.Globals.Keys)
+			{
+				var fieldValue = runtime.Globals[fieldName];
+				if (!allowedFunctions.Contains(fieldName.ToString()))
+					runtime.Globals[fieldName] = null;
```

Comment:
should we throw errors here? Or at least log them?

Reply:
Your comment confused me. I thought this was removing functions from the script, but it removes them from the Lua runtime. So no to both.

Reply:
makes sense